tests/cases/compiler/indexSignatureAndMappedType.ts

```diff
@@ -0,0 +1,35 @@
+// @strict: true
+// @declaration: true
+
+// A mapped type { [P in K]: X }, where K is a generic type, is related to
+// { [key: string]: Y } if X is related to Y.
+
+function f1<T, K extends string>(x: { [key: string]: T }, y: Record<K, T>) {
+    x = y;
+    y = x;  // Error
+}
+
+function f2<T, K extends string>(x: { [key: string]: T }, y: Record<string, T>) {

```

Comment:
To me it appears to be unused and unnecessary for the test. @ahejlsberg, is the type parameter `K` needed for this test?